resolver/GlobalPass.cc

```diff
@@ -388,6 +545,14 @@ void Resolver::finalizeSymbols(core::GlobalState &gs) {
 
     computeLinearization(gs);
 
+    // Compute the lineazrize list of all required ancestors for each class or module
+    RequiredAncestorsLinearization requiredAncestorsLinearization(gs);
```

Comment:
Why is this a class instead of a function? It looks like the only benefit is to avoid passing around `gs` to every helper function. We prefer threading that around explicitly (there is pretty much no class that takes a `GlobalState` and holds onto it).

Reply:
I need it to hold a local state for the linearization cache: https://github.com/sorbet/sorbet/pull/3409/files/5fd178a1dfa0c3f1784a66a46adf9573ab32d6ff#diff-9645170eeb3775ef8bd3dac5fe8375b9R378-R379

I can add all of this in the `Resolver` if you prefer?

Reply:
I see. In that case, I'd prefer keeping it a class, but still taking the `GlobalState &` argument by manually threading it through all the call sites.